sample/sample-with-di/app/src/main/java/com/dropbox/kaiken/sample/advancedsample/app/AppServices.kt

```diff
@@ -1,46 +1,40 @@
 package com.dropbox.kaiken.sample.advancedsample.app
 
-import com.dropbox.kaiken.sample.advancedsample.helloworldfeature.HelloWorldDependencies
 import com.dropbox.kaiken.sample.advancedsample.helloworldfeature.HelloWorldMessageProvider
 import com.dropbox.kaiken.sample.advancedsample.helloworldfeature.RealTimeMessageProvider
 import com.dropbox.kaiken.sample.advancedsample.helloworldfeature.RealWorldMessageProvider
 import com.dropbox.kaiken.sample.advancedsample.helloworldfeature.TimeMessageProvider
-import com.dropbox.kaiken.scoping.AppServices
+import com.dropbox.kaiken.scoping.AppScope
+import com.dropbox.kaiken.scoping.SingleIn
 import com.dropbox.kaiken.scoping.TeardownHelper
-import dagger.Component
+import com.squareup.anvil.annotations.ContributesBinding
+import com.squareup.anvil.annotations.ContributesTo
 import dagger.Module
 import dagger.Provides
+import kotlinx.coroutines.flow.MutableSharedFlow
+import javax.inject.Inject
 
-annotation class AppScope
-
-@Component(
-    modules = [AppServicesModule::class]
-)
-@AppScope
-abstract class AdvancedKaikenSampleAppServices : AppServices, HelloWorldDependencies, CoreAppServices {
-
-    @Component.Factory
-    interface Factory {
-        fun create(): AdvancedKaikenSampleAppServices
-    }
-
-    override fun getTeardownHelper(): TeardownHelper = NOOP_TEARDOWNHELPER
-}
 
 @Module
+@ContributesTo(AppScope::class)
 object AppServicesModule {
     @Provides
-    @AppScope
+    @SingleIn(AppScope::class)
```

Comment:
I found this type of singleton annotation to be simpler to use since we can reuse the same key for singleton and anvil scope